src/client/ui/panels/PropertiesPanelContainer.js

```diff
@@ -158,12 +159,21 @@ class PropertiesPanelContainer extends Component {
     );
   };
 
-  getExtras(prop) {
+  getExtras(prop, component) {
     switch (prop.type) {
       case types.number:
         return { min: prop.min, max: prop.max };
-      case types.file:
-        return { openFileDialog: this.props.openFileDialog, filters: prop.filters };
+      case types.file: {
+        const isValid =
+          component._object && component._object.resourcesValidation
+            ? component._object.resourcesValidation[prop.name]
+            : true;
```

Comment:
This feels wrong. You're mixing meanings here. The extras are supposed to be based off the prop schema, not the component instance state itself, and it doesn't seem right to store validation on the object anyway, as I mentioned above.